modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm

```diff
@@ -0,0 +1,392 @@
+	var/delay_to_add = 2
+
+/obj/item/microfusion_gun_attachment/repeater/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil += recoil_to_add
+	microfusion_gun.burst_size += burst_to_add
+	microfusion_gun.fire_delay += delay_to_add
+	microfusion_gun.spread += spread_to_add
+
+/obj/item/microfusion_gun_attachment/repeater/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil -= recoil_to_add
+	microfusion_gun.burst_size -= burst_to_add
+	microfusion_gun.fire_delay -= delay_to_add
+	microfusion_gun.spread -= spread_to_add
+
+/*
+X-RAY ATTACHMENT
+
+The gun can fire X-RAY shots.
+*/
+/obj/item/microfusion_gun_attachment/xray
+	name = "quantum phase inverter array" //Yes quantum makes things sound cooler.
+	desc = "Experimental technology that inverts the central phase emitter causing the wave frequency to shift into X-ray. CAUTION: Phase emitter heats up very quickly."
+	icon_state = "attachment_xray"
+	slot = GUN_SLOT_BARREL
+	attachment_overlay_icon_state = "attachment_xray"
+	heat_addition = 90
+	power_usage = 50
+
+/obj/item/microfusion_gun_attachment/xray/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.fire_sound = 'modular_skyrat/modules/microfusion/sound/incinerate.ogg'
+
+/obj/item/microfusion_gun_attachment/xray/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.fire_sound = microfusion_gun.chambered?.fire_sound
+
+/obj/item/microfusion_gun_attachment/xray/process_fire(obj/item/gun/microfusion/microfusion_gun, obj/item/ammo_casing/chambered)
+	. = ..()
+	chambered.loaded_projectile.icon_state = "laser_greyscale"
+	chambered.loaded_projectile.color = COLOR_GREEN
+	chambered.loaded_projectile.light_color = COLOR_GREEN
+	chambered.loaded_projectile.projectile_piercing = PASSCLOSEDTURF|PASSGRILLE|PASSGLASS
+
+/*
+GRIP ATTACHMENT
+
+Greatly reduces recoil and spread.
+*/
+/obj/item/microfusion_gun_attachment/grip
+	name = "grip attachment"
+	desc = "A simple grip that increases accuracy."
+	icon_state = "attachment_grip"
+	attachment_overlay_icon_state = "attachment_grip"
+	slot = GUN_SLOT_UNDERBARREL
+	/// How much recoil are we removing?
+	var/recoil_to_remove = 1
+	/// How much spread are we removing?
+	var/spread_to_remove = 10
+
+/obj/item/microfusion_gun_attachment/grip/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil -= recoil_to_remove
+	microfusion_gun.spread -= spread_to_remove
+
+/obj/item/microfusion_gun_attachment/grip/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil += recoil_to_remove
+	microfusion_gun.spread += spread_to_remove
+
+/*
+HEATSINK ATTACHMENT
+
+"Greatly increases the phase emitter cooling rate."
+*/
+/obj/item/microfusion_gun_attachment/heatsink
+	name = "phase emitter heatsink"
+	desc = "Greatly increases the phase emitter cooling rate."
```

Comment:
desc = "A simple attachment mounted under the barrel, offering twice the heat dissipation capacity of the weapon's inbuilt sink."